src/rust/ide/view/src/searcher/icons.rs

```diff
@@ -0,0 +1,387 @@
+//! All icons that are used in the searcher.
+
+use ensogl::prelude::*;
+
+use ensogl::application::Application;
+use ensogl::data::color;
+use ensogl::display::DomSymbol;
+use ensogl::display::navigation::navigator::Navigator;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::shape::*;
+use ensogl::system::web;
+use ensogl_theme::application::searcher::icons as theme;
+use ensogl::system::web::StyleSetter;
+use std::f32::consts::PI;
+use wasm_bindgen::prelude::*;
+
+/// The width and height of all icons.
+const ICON_SIZE : f32 = 16.0;
+
+// Due to a rendering error, shapes appear too big when the camera is zoomed in very closely.
+// (Documented here: https://github.com/enso-org/ide/issues/1698)
+// To compensate for this, we apply `.shrink(SHRINK_AMOUNT)` to all icons in this file. By default,
+// `SHRINK_AMOUNT` should be set to 0.0 to make the icons appear right on the default zoom level.
+// When examining the icons closely, with a strong zoom, `SHRINK_AMOUNT` should be set to 0.4 to
+// make the icons apear right on that zoom level.
+const SHRINK_AMOUNT : f32 = 0.0;
+// const SHRINK_AMOUNT : f32 = 0.38;
```

Comment:
I was already writing comment about it somewhere. I do not understand it. First of all it is set to 0.0. What does it mean that you want to watch icons with close zoom? With a camera zoom? What is the idea this variable should serve for? Do you want it to be ever changed? I do not understand the purpose of this tbh. And also commented code.

Reply:
I reformulated it again. Is this clearer?

> Due to a rendering error, shapes appear too big when the camera is zoomed in very closely.
(Documented here: https://github.com/enso-org/ide/issues/1698)
In the user interface, this is not a big problem, since icons are usually shown at lower zoom levels. But it is a problem during development of icons when it becomes necessary to inspect them closely. In those situations, one can apply `.shrink(0.38)` to shapes to compensate for the bug make them appear at the correct size while the camera is zoomed in. But that work-around will make them appear too thin on the default zoom level.

> To make it easy to turn this shrinking on and off before and after working on icons, we define the constant `SHRINK_AMOUNT` and apply `.shrink(SHRINK_AMOUNT)` to all icons. In every commit, `SHRINK_AMOUNT` should be set to 0.0 to make icons look best in the user interface. But during work on the icons, it can temporarily be set to 0.38.

Reply:
Yes, now I understand that it is for development purposes only. I would start the whole docs with "For development purposes only." to make it even more clear!